### DISCUSSION TITLE
Custom theme: styles not applying

### DISCUSSION THREAD

**Discussion by keH1** [Power-Components/livewire-powergrid#1477] [Q&A] [answered]
Hello everyone, I've encountered a problem and can't figure out what's wrong. I'm trying to customize a table as instructed in the customization guide, and created my own class `'theme' => \App\Themes\TableGridTheme::class,` and registered it in the config. Like in instruction: https://livewire-powergrid.com/table/custom-theme.html

I inherited from the Tailwind theme, refreshed the page - nothing works. Added to the search box a class "eewfewfwef", deleted all the others - refreshed the page - nothing.

I reverted to the classic theme `'theme' => \PowerComponents\LivewirePowerGrid\Themes\Tailwind::class,` went to the path `vendor/power-components/livewire-powergrid/src/Themes/Tailwind.php` There, I added this class instead of others `eewfewfwef` and everything worked!

What's the problem? Why can't it override the classes that I set in my own theme? Moreover, if I change the theme name, everything crashes with an exception that the theme cannot be found...

Here is what I'm customizing: 
```php
public function searchBox(): SearchBox
    {
        return Theme::searchBox()
            ->input('eewfewfwef')
            ->iconClose('text-pg-primary-400 dark:text-pg-primary-200')
            ->iconSearch('text-pg-primary-300 mr-2 w-5 h-5 dark:text-pg-primary-200');
    }
```
and if it's in my class - nothing happened.
<img width="1644" alt="image" src="https://github.com/Power-Components/livewire-powergrid/assets/652726/8955869a-2ede-4cb6-b828-16402d6cca6f">

changing to the default theme and customize in the vendor folder - works great
<img width="1644" alt="image" src="https://github.com/Power-Components/livewire-powergrid/assets/652726/a595f40b-cf17-442c-8ef0-c8e7fa4aeee6">

I did everything as in the instructions. Any ideas? 

Laravel 10.48 and Last PowerGrids

**Comment by marineusde** [comment, marked as answer]
I cant reproduce it. I've created a custom theme:

```
class CustomTheme extends Tailwind
{
    public function searchBox(): \PowerComponents\LivewirePowerGrid\Themes\Components\SearchBox
    {
        return Theme::searchBox()
            ->input('fancy_input')
            ->iconClose('fancy_icon_close')
            ->iconSearch('fancy_search');
    }
}
```

configure it in the livewire-powergrid.php:
```
  'theme' => \App\PowergridThemes\CustomTheme::class,
```

Result:
![Bildschirmfoto vom 2024-04-03 14-31-08](https://github.com/Power-Components/livewire-powergrid/assets/105713989/c34728cd-076a-41cd-92b6-54c697d56463)

I tryed it with bootstrap, works too.

Maybe you can try to delete all caches:

```
composer dumpautoload --optimize
php artisan optimize
php artisan cache:clear
php artisan route:clear
php artisan config:clear
php artisan view:clear
php artisan config:clear
php artisan clear-compiled
```

**Reply by keH1** [reply]
Yes, problem was been in cache, thanks!
